src/OpenTelemetry.ResourceDetectors.Container/ContainerResourceDetector.cs

```diff
@@ -18,6 +21,13 @@ public class ContainerResourceDetector : IResourceDetector
     private const string Filepath = "/proc/self/cgroup";
     private const string FilepathV2 = "/proc/self/mountinfo";
     private const string Hostname = "hostname";
+    private const string K8sServiceHostKey = "KUBERNETES_SERVICE_HOST";
+    private const string K8sServicePortKey = "KUBERNETES_SERVICE_PORT_HTTPS";
+    private const string K8sNamespaceKey = "KUBERNETES_NAMESPACE";
```

Comment:
Is this environment variable and `KUBERNETES_CONTAINER_NAME` available in all Kubernetes environments?

Reply:
From what I know, `KUBERNETES_NAMESPACE` and `KUBERNETES_CONTAINER_NAME` are not available by default in Kubernetes. The user will have to provide them using the [downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/).